third_party/clang/google_format.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+#
+# Copyright 2016 The Closure Rules Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# clang-format's intended use case is to automatically reformat source files
+# without need for human intervention. Therefore the primary means of source
+# code formatting should be performed automatically from the editor. This
+# script checks returns success if no reformatting is required. However, if
+# reformatting is required an error is returned. In this case, the source file
+# is _not_ reformatted. Instead, the user must invoke clang-format (possibly
+# via their editor) in order for the source input file to be formatted.
+
+set -e
+set -u
+
+$1 -style=Google -output-replacements-xml "${@:3}" | \
```

Comment:
I would recommend removing this from this CL since it seems like a utility that would be used in the background by a rule. Also the third_party directory isn't a good place for scripts we write ourselves.

If you want, consider adding a script so that I can say:

```sh
bazel build @io_bazel_rules_closure//closure/style:fix
./bazel-bin/external/io_bazel_rules_closure/closure/style/fix $(find mydir -name \*.js)
```

But that might be better for a follow-up pull request. I'm happy with this one just making clang-format available.